solr/core/src/java/org/apache/solr/servlet/SolrRequestParsers.java

```diff
@@ -571,59 +552,99 @@ public SolrParams parseParamsAndFillStreams(
     }
   }
 
-
-
   /**
    * Extract Multipart streams
    */
   static class MultipartRequestParser implements SolrRequestParser {
-    private final int uploadLimitKB;
-    private DiskFileItemFactory factory = new DiskFileItemFactory();
-    
-    public MultipartRequestParser(int limit) {
-      uploadLimitKB = limit;
+    private final MultipartConfigElement multipartConfigElement;
 
-      // Set factory constraints
-      FileCleaningTracker fct = fileCleaningTracker;
-      if (fct != null) {
-        factory.setFileCleaningTracker(fileCleaningTracker);
-      }
-      // TODO - configure factory.setSizeThreshold(yourMaxMemorySize);
-      // TODO - configure factory.setRepository(yourTempDirectory);
+    public MultipartRequestParser(int uploadLimitKB) {
+      multipartConfigElement = new MultipartConfigElement(
+          null, // temp dir (null=default)
+          -1, // maxFileSize  (-1=none)
+          uploadLimitKB * 1024, // maxRequestSize
+          1 * 1024 * 1024 ); // fileSizeThreshold after which will go to disk
```

Comment:
```suggestion
          100 * 1024 ); // fileSizeThreshold after which will go to disk
```
Lowering to reduce DoS risk